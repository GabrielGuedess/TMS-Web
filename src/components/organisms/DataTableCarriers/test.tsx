import { render } from '@testing-library/react';

import { DataTableCarriers } from '.';

jest.mock('lottie-react', () => ({
  __esModule: true,
  default: () => <div data-testid="Mock Lottie" />,
}));

jest.mock('ag-grid-react', () => ({
  AgGridReact: () => <div data-testid="Mock AG Grid" />,
}));

describe('<DataTableCarriers />', () => {
  it('should render the heading', () => {
    const { container } = render(<DataTableCarriers />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
