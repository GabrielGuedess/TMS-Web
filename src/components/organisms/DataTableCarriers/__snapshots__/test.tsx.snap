// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTableCollaborators /> should render the heading 1`] = `
<div
  class="relative overflow-x-auto shadow-default sm:rounded"
>
  <div
    class="flex flex-col"
  >
    <div
      class="relative flex flex-wrap-reverse items-center justify-between gap-2 bg-white-lilac-50 p-4 transition-all dark:bg-smoke-950 md:flex-row md:flex-wrap md:space-y-0"
    >
      <div
        class="flex items-center gap-2"
      >
        <button
          aria-autocomplete="none"
          aria-controls="radix-:r0:"
          aria-expanded="false"
          class="group relative flex items-center justify-between rounded-lg border-2 border-zumthor-100 bg-white-lilac-50 px-3 py-1.5 text-sm font-medium text-comet-500 transition-all hover:bg-porcelain-50 focus:outline-none dark:border-shark-950 dark:bg-shark-950 dark:text-dark-300 dark:hover:opacity-60 focus:border-primary-300 dark:focus:border-primary-400 dark:data-[state="open"]:border-primary-400 data-[state="open"]:border-primary-300 focus:placeholder:opacity-100"
          data-state="closed"
          dir="ltr"
          role="combobox"
          type="button"
        >
          <span
            style="pointer-events: none;"
          >
            25
          </span>
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="ms-2.5 size-2.5"
              fill="none"
              viewBox="0 0 10 6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m1 1 4 4 4-4"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </button>
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="flex items-center rounded-lg border border-zumthor-100 bg-white-lilac-50 px-3 py-1.5 text-sm font-medium text-comet-500 transition-all hover:bg-porcelain-50 focus:outline-none focus:ring-4 focus:ring-gray-200 dark:border-shark-950 dark:bg-shark-950 dark:text-dark-300 dark:hover:opacity-60 dark:focus:ring-gray-700 opacity-0 pointer-events-none"
          data-state="closed"
          type="button"
        >
          <span>
            Linhas alteradas:
             
            <span
              class="text-primary-400"
            >
              0
            </span>
          </span>
        </button>
        <button
          aria-controls="radix-:r4:"
          aria-expanded="false"
          aria-haspopup="dialog"
          aria-label="Clear Edit"
          class="p-1 text-primary-400 outline-primary-400 transition-all hover:rotate-180 focus:rotate-180 opacity-0 pointer-events-none"
          data-state="closed"
          type="button"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 22C17.5 22 22 17.5 22 12C22 6.5 17.5 2 12 2C6.5 2 2 6.5 2 12C2 17.5 6.5 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M9.16998 14.83L14.83 9.17"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M14.83 14.83L9.16998 9.17"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
      </div>
      <div
        class="relative flex flex-1 gap-3 md:flex-initial"
      >
        <a
          href="/dashboard/collaborators/new"
        >
          <button
            class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90"
            type="button"
          >
            <div
              aria-label="Button Background"
              class="absolute size-full rounded-md opacity-[16%] bg-primary-400"
            />
            <div
              class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-primary-400"
            >
              <span
                class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-[0.9375rem]"
              >
                Adicionar Usu√°rio
              </span>
            </div>
          </button>
        </a>
        <form
          class="relative flex flex-1 md:flex-initial"
        >
          <button
            aria-label="search"
            class="absolute inset-y-0 start-0 flex cursor-pointer items-center ps-3 text-comet-500 transition-all hover:text-primary-400 focus:text-primary-400 dark:text-dark-300 dark:hover:text-primary-400 dark:focus:text-primary-400"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="size-4 transition-all"
              fill="none"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <input
            class="flex flex-1 rounded-lg border border-zumthor-100 bg-gray-50 p-2 ps-10 text-sm text-shark-900 outline-primary-400 transition-all focus:border-primary-400 focus:ring-primary-400 dark:border-shark-950 dark:bg-shark-950 dark:text-white-lilac-50 dark:placeholder:text-dark-300 dark:focus:border-primary-400 dark:focus:ring-primary-400 md:w-80 md:flex-initial"
            name="search"
            placeholder="Pesquisar"
            type="text"
          />
        </form>
      </div>
    </div>
    <div
      class="flex animate-pulse flex-col gap-4 pb-4"
    >
      <div
        class="flex h-8 w-full rounded bg-porcelain-50 transition-all dark:bg-shark-950"
      />
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-4 gap-4 px-4"
      >
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
        <div
          class="mb-4 flex h-4 flex-1 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
      <div
        class="flex h-16 w-full items-center rounded bg-porcelain-50 px-4 transition-all dark:bg-shark-950"
      >
        <div
          class="h-5 w-1/5 rounded bg-dark-100 transition-all dark:bg-shark-950"
        />
      </div>
    </div>
  </div>
</div>
`;
